src/main/controllers/ChallengeController.kt

```diff
@@ -41,7 +41,19 @@ class ChallengeController: DefaultController<Challenge>(), RestController<Challe
         throw NotImplementedError()
     }
 
-    fun share(user: UserAccount, request: Request): SOAResult<Challenge> {
-        throw NotImplementedError()
+    fun share(user: UserAccount, request: Request): SOAResult<Transaction> {
```

Comment:
so this should return a Pair<Transaction, NEwUSerAccount> in case it shares with a new user

Reply:
and need to test the case of a new user